src/test_utils/mod.rs

```diff
@@ -110,28 +115,26 @@ mod tests {
 
     #[test]
     fn test_delete_attribute() -> Result<(), Error> {
-        let mut policy = policy()?;
-        let cover_crypt = Covercrypt::default();
-        let (mut msk, _) = cover_crypt.setup()?;
-        let mpk = cover_crypt.update_master_keys(&policy, &mut msk)?;
+        let (mut msk, mpk, cover_crypt) = msk()?;
 
         // New user secret key
         let decryption_policy = AccessPolicy::parse(
             "Security Level::Top Secret && (Department::FIN || Department::HR)",
         )?;
         let mut top_secret_fin_usk =
-            cover_crypt.generate_user_secret_key(&mut msk, &decryption_policy, &policy)?;
+            cover_crypt.generate_user_secret_key(&mut msk, &decryption_policy)?;
 
         // Encrypt
         let top_secret_ap = AccessPolicy::parse("Security Level::Top Secret && Department::FIN")?;
         let (_, encrypted_header) =
-            EncryptedHeader::generate(&cover_crypt, &policy, &mpk, &top_secret_ap, None, None)?;
+            EncryptedHeader::generate(&cover_crypt, &mpk, &top_secret_ap, None, None)?;
 
         // remove the FIN department
-        policy.remove_attribute(&Attribute::new("Department", "FIN"))?;
+        msk.policy
+            .remove_attribute(&Attribute::new("Department", "FIN"))?;
```

Comment:
Okay, I may have been hasty in my comment advicing not to expose the policy publicly. I a first time, we can publicly expose the policy from the MSK *only*. In a second time we may re-export all policy-related methods through the MSK, calling `update` each time. This will be costly since `update` is rather blind (it goes through all the space to check for modifications). In a third time, we will modify `update` so that it focuses on the narrow sub-space in which modifications have occurred.

Reply:
The target for the release is the complete internalization of the policy (the second time).